modules/pdf_generator/report/procedures.php

```diff
@@ -0,0 +1,207 @@
+  );
+
+  while ($row_pdf = sqlFetchArray($res)) {
+    $report_id = empty($row_pdf['procedure_report_id']) ? 0 : ($row_pdf['procedure_report_id'] + 0);
+
+    $query = "SELECT " .
+      "ps.result_code, ps.result_text, ps.abnormal, ps.result, ps.range, " .
+      "ps.result_status, ps.facility, ps.units, ps.comments, ps.document_id, ps.date " .
+      "FROM procedure_result AS ps " .
+      "WHERE ps.procedure_report_id = ? " .
+      "ORDER BY ps.result_code, ps.procedure_result_id";
+
+    $rres = sqlStatement($query, array($report_id));
+
+    if ($finals_only) {
+      // We are consolidating reports.
+      if (sqlNumRows($rres)) {
+        $rrowsets = array();
+        // First pass creates a $rrowsets[$key] for each unique result code in *this* report, with
+        // the value being an array of the corresponding result rows. This caters to multiple
+        // occurrences of the same result code in the same report.
+        while ($rrow_pdf = sqlFetchArray($rres)) {
+          $result_code = empty($rrow_pdf['result_code']) ? '' : $rrow_pdf['result_code'];
+          $key = sprintf('%05d/', $row_pdf['procedure_order_seq']) . $result_code;
+          if (!isset($rrowsets[$key])) $rrowsets[$key] = array();
+          $rrowsets[$key][] = $rrow_pdf;
+        }
+        // Second pass builds onto the array of final results for *all* reports, where each final
+        // result for a given result code is its *array* of result rows from *one* of the reports.
+        foreach ($rrowsets as $key => $rrowset) {
+          // When two reports have the same date, use the result date to decide which is "latest".
+          if (isset($finals[$key]) &&
+            $row_pdf['date_report'] == $finals[$key][0]['date_report'] &&
+            !empty($rrow_pdf['date']) && !empty($finals[$key][1]['date']) &&
+            $rrow_pdf['date'] < $finals[$key][1]['date'])
+          {
+            $finals[$key][2] = true;
+            continue;
+          }
+          $finals[$key] = array($row_pdf, $rrowset, isset($finals[$key]));
+        }
+      }
+      else {
+        // We have no results for this report.
+        $key = sprintf('%05d/', $row_pdf['procedure_order_seq']);
+        $finals[$key] = array($row_pdf, array($empty_results), false);
+      }
+    }
+    else {
+      // We are showing all results for all reports.
+      if (sqlNumRows($rres)) {
+        while ($rrow = sqlFetchArray($rres)) {
+          generate_result_row_pdf($ctx_pdf, $row_pdf, $rrow, false);
+        }
+      }
+      else {
+        generate_result_row_pdf($ctx_pdf, $row_pdf, $empty_results, false);
+      }
+    }
+}
+
+  if ($finals_only) {
+    ksort($finals);
+    foreach ($finals as $final) {
+      foreach ($final[1] as $rrow) {
+        generate_result_row_pdf($ctx_pdf, $final[0], $rrow_pdf, $final[2]);
+      }
+    }
+  }
+  $content_pro .= "</table><br><br>";
+}
+
+if($i != 1)
+{
+  $pdf->WriteHTML($content_pro, true, false, false, false, '');
```

Comment:
Where do I find that function?

Reply:
If you are talking about WriteHTML method then its TCPDF's default method and working on the object $pdf declared in selector.php. Is the code working fine?